src/calendar.h

```diff
@@ -342,4 +342,286 @@ class calendar
         static season_type initial_season;
 };
 
+class time_duration;
```

Comment:
I think this is extensive enough to merit its own header, even if calendar.h just ends up including it.
It looks like quite a few files in this PR are adding a dependency on calendar.h when they only need time.h